standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/AbstractThriftHiveMetastore.java

```diff
@@ -0,0 +1,1610 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore;
+
+import com.facebook.fb303.fb_status;
+import org.apache.hadoop.hive.metastore.api.*;
+import org.apache.hadoop.hive.metastore.api.Package;
+import org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore.Iface;
+import org.apache.thrift.TException;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This abstract class can be extended by any remote cache that implements HMS APIs.
+ * The idea behind introducing this abstract class is not to break the build of remote cache,
+ * whenever we add new HMS APIs.
+ */
+public abstract class AbstractThriftHiveMetastore implements Iface {
```

Comment:
Just thinking out loud here.
The name AbstractThriftHiveMetastore suggests thats it is is an abstract class which is used internally by Hive and also extensible by other implementations. The fact that Hive doesn't use it at all, it feels like a Dummy implementation that should be named DummyThriftHiveMetastore or something like that. But then, it would be odd for new concrete implementations to extend a Dummy class.

Perhaps something like PluggableThriftHiveMetastore might be the right name as it indicates it used to plugin new implementations.

+1 on the change for me, I will leave it to you on the name.

Reply:
Concept of an interface itself makes all these APIs pluggable with your own implementation, so not sure if that's what this class brings in extra. @vihangk1 Do you have any other suggestions for the class name ? @nrg4878 This can be extended in Hive as well, for example by CachedStore in future.